### PR TITLE
Ask buildtools to also restore ILAsm

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Will only do something after dotnet/buildtools#1806 is picked up. We should start with having this restored before we start using it to make sure that nobody has tools downloaded/initialized without ILAsm.

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet/dncenghot All CoreCLR OSX CI jobs are timing out on what appears to git hang:

```
02:12:56 Commit message: "Merge 9b831c077c7a927315a2abe2eb4747ecfa256fc5 into b9affb49f79ed8319923c2f5b07b4aeeb04efe36"
02:12:56  > git config core.sparsecheckout # timeout=10
02:12:56  > git checkout -f 32efdda2fa45fb0a52c8c8e9765c9c1cf5ddc245
02:13:01 First time build. Skipping changelog.
04:13:01 
Build timed out (after 120 minutes)
```

**Comment by karajas** [comment]
Filed https://github.com/dotnet/core-eng/issues/2143 to track this.

**Comment by karajas** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by MichalStrehovsky** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test

**Comment by MichalStrehovsky** [comment]
@dotnet-bot test Perf Build and Test

**Comment by jkotas** [comment]
@MichalStrehovsky Perf Build and Test tends to fail for random reasons, and it takes forever to finish when it succeeds. You do not need to block on it.

cc @valenis @brianrob

**Comment by MichalStrehovsky** [comment]
> Perf Build and Test tends to fail for random reasons

I just wanted to be extra careful, because build changes tend to break unexpected stuff, but not blocking on this sounds good. Thanks!

File: ILAsmVersion.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+2.1.0-preview1-25916-01
```
Comment (weshaggard) [inline ILAsmVersion.txt]: You should consider adding this to the auto-update list in dependencies.props. Follow the similar pattern that we use for updating the BuildToolsVersion.txt file at https://github.com/dotnet/coreclr/blob/master/dependencies.props#L97.

**Pull request merged**